.github/workflows/test.yaml

```diff
@@ -9,11 +9,12 @@ on:
 jobs:
   build_and_test:
     name: Build & Test
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
```

Comment:
do we really want to run all tests on all platforms?

Reply:
These tests are faster than the e2e tests, so they shouldn't slow anything down. But I'd be fine to execute a subset only too.

Reply:
I mean, I know we don't pay for any of this, but it seems a little wasteful when we would get the same info from the e2e

Reply:
removed